examples/Validate/check_cpu_memory.yaml

```diff
@@ -6,33 +6,33 @@ spec:
   rules:
   - name: check-defined
     resource:
-      kinds: 
+      kinds:
       - Deployment
     validate:
-      message: "Resource requests and limits are required for CPU and memory"
+      message: "Resource limits are required for CPU and memory"
       pattern:
-      spec:
-        containers:
-        - name: "*"
-          resources:
-            limits:
-              memory: "?"
-              cpu: "?"
-            requests:
-              memory: "?"
-              cpu: "?"
-  - name: check-memory-in-range
+        spec:
+          template:
+            spec:
+              containers:
+              - name: "*"
+                resources:
+                  limits:
+                    memory: "*"
```

Comment:
Shouldn't this be "?*" or "*?"....?

"*" will allow a empty value, which is not the intent of the policy.

Reply:
If we use the pattern: “?”, then if you specify a value with more than one character, the resource will fail validation.
If we use the pattern: “*”, validation checks only the presence of this field, and if it is empty, the Kubernetes itself will generate an error when creating the resource "unknown object type "nil" in Deployment.spec.template.spec.containers[0].resources.limits.cpu;"
If we use the pattern: “?*”/"*?", then we will be sure that the field is specified and it is not empty. This option is more reliable, we can use it.